pkg/validation/apiextensions/v1/composition/readinessChecks.go

```diff
@@ -83,7 +83,7 @@ func getReadinessCheckExpectedType(r v1.ReadinessCheck) xpschema.KnownJSONType {
 		matchType = xpschema.KnownJSONTypeString
 	case v1.ReadinessCheckTypeMatchInteger:
 		matchType = xpschema.KnownJSONTypeInteger
-	case v1.ReadinessCheckTypeNone, v1.ReadinessCheckTypeNonEmpty, v1.ReadinessCheckTypeMatchCondition:
+	case v1.ReadinessCheckTypeNone, v1.ReadinessCheckTypeNonEmpty, v1.ReadinessCheckTypeMatchCondition, v1.ReadinessCheckTypeMatchTrue, v1.ReadinessCheckTypeMatchFalse:
```

Comment:
here it should actually return `xpschema.KnownJSONTypeBoolean` for `MatchTrue` and `MatchFalse`

Reply:
Right, thanks! Fixed. Guess it was working before because we only compare types if this func returns a non-empty `xpschema.KnownJSONType`